main.go

```diff
@@ -324,3 +353,238 @@ func printList() {
 	}
 	fmt.Println(b.String())
 }
+
+// locateFile locates a file named file, or a directory if name is empty, and returns it's full path
+// It first tries in the paths given by the dirs, and then a system lookup
+func locateFile(file string, dirs []string) string {
+	var res string
+	for _, dir := range dirs {
+		if dir != "" {
+			fi, err := os.Stat(filepath.Join(dir, file))
+			if err == nil && ((file == "" && fi.IsDir()) || (file != "" && fi.Mode().IsRegular())) {
+				return filepath.Join(dir, file)
+			}
+		}
+	}
+	if file != "" && res == "" {
+		p, _ := exec.LookPath(file)
+		if p != "" {
+			return p
+		}
+	}
+	return ""
+}
+
+// getBPFObject finds or builds ebpf object file and returns it's path
+func getBPFObject() (string, error) {
+	bpfObjFileName := "tracee.bpf.o"
+	exePath, err := os.Executable()
+	if err != nil {
+		return "", err
+	}
+	//locations to search for the bpf file, in the following order
+	searchPaths := []string{
+		os.Getenv("TRACEE_BPF_FILE"),
+		filepath.Dir(exePath),
+		traceeInstallPath,
+	}
+	bpfObjFilePath := locateFile(bpfObjFileName, searchPaths)
+	if bpfObjFilePath != "" && debug {
+		fmt.Printf("found bpf object file at: %s\n", bpfObjFilePath)
+	}
+	if bpfObjFilePath == "" {
+		if debug {
+			fmt.Printf("could not find tracee bpf object file in any of %v. attempting to build it\n", searchPaths)
+		}
+		madeBPFObjFilePath := filepath.Join(traceeInstallPath, bpfObjFileName)
+		err = makeBPFObject(madeBPFObjFilePath)
+		if err != nil {
+			return "", err
+		}
+		if debug {
+			fmt.Printf("succesfully built ebpf obj file into: %s\n", madeBPFObjFilePath)
+		}
+		bpfObjFilePath = madeBPFObjFilePath
+	}
+
+	return bpfObjFilePath, nil
+}
+
+// getBPFSource puts the ebpf source in a file under dir and returns the path to the file
+func getBPFSourceFile(dir string) string {
+	if ebpfProgramB64Injected == "" {
+		return ""
+	}
+	progReader := base64.NewDecoder(base64.RawStdEncoding, strings.NewReader(ebpfProgramB64Injected))
+	progFilePath := filepath.Join(dir, "tracee.bpf.c")
+	os.Remove(progFilePath)
+	progFile, _ := os.Create(progFilePath)
+	defer progFile.Close()
+	io.Copy(progFile, progReader)
+	return progFilePath
+}
+
+// makeBPFObject builds the ebpf object from source code into the provided path
+func makeBPFObject(outFile string) error {
+	dir, err := ioutil.TempDir("", "tracee-make")
+	if err != nil {
+		return err
+	}
+	if debug {
+		fmt.Printf("building bpf object in: %s\n", dir)
+	} else {
+		defer os.RemoveAll(dir)
+	}
+	objFile := filepath.Join(dir, "tracee.bpf.o")
+	bpfSrc := getBPFSourceFile(dir)
+	if bpfSrc == "" {
+		return fmt.Errorf("could not get bpf source file")
+	}
+	clang := locateFile("clang", []string{os.Getenv("CLANG")})
```

Comment:
Not sure this is a good idea (security wise).
Tracee is running with root privileges, we have to be sure we are not executing a malicious executables that was found during search.

Reply:
Maybe we can drop privileges before execution of these files

Reply:
this is a good feedback, but I won't have time to implement it soon, maybe end of week or next week. Do you want to wait for it, or merge it and fix in a followup PR?

Reply:
we need to "install" as root anyway in order to copy the files into a permanent location

Reply:
Doing make doesn't require root privileges.
Tracee will not start without root privileges.
We can install not as root, and save to some temp dir.
I think this is better to put the object file in a dir that is deleted after reboot anyways. This way, you can be sure that the bpf code was compiled with the correct kernel version. If the user will want to install it to a different location that requires root privileges, he will need to do this explicitly